test_util/aws.py

```diff
@@ -170,30 +195,14 @@ def empty_and_delete_s3_bucket_from_stack(self):
         bucket.delete()
         log.info('Delete successfully triggered for {}'.format(self.stack.stack_name))
 
-    def get_master_ips(self, state_list=None):
-        instances = self.get_group_instances(['MasterServerGroup'], state_list)
-        return instances_to_hosts(instances)
-
-    def get_public_agent_ips(self, state_list=None):
-        instances = self.get_group_instances(['PublicSlaveServerGroup'], state_list)
-        return instances_to_hosts(instances)
+    def get_master_ips(self):
+        return instances_to_hosts(self.get_auto_scaling_instances('MasterServerGroup'))
 
-    def get_private_agent_ips(self, state_list=None):
-        instances = self.get_group_instances(['SlaveServerGroup'], state_list)
-        return instances_to_hosts(instances)
+    def get_public_agent_ips(self):
+        return instances_to_hosts(self.get_auto_scaling_instances('PublicSlaveServerGroup'))
```

Comment:
see above; how do we decide when to refresh and pickup terminated or added nodes?